src/protocols/aeternity/libs/AeternityAdapter.ts

```diff
@@ -334,35 +347,83 @@ export class AeternityAdapter extends BaseProtocolAdapter {
     }
   }
 
-  override async fetchTransactions(
-    address: string,
-    nextPage: string | null,
+  override async fetchAccountTransactions(
+    address: AccountAddress,
+    { nextPageUrl }: ITransactionApiPaginationParams = {},
   ): Promise<IFetchTransactionResult> {
-    const { getAeSdk } = useAeSdk();
-    await getAeSdk(); // Ensure the `nodeNetworkId` is established
-
-    if (typeof nextPage !== 'string') {
-      return {
-        regularTransactions: [],
-        nextPageParams: null,
-      };
-    }
+    let paginationParams: ITransactionApiPaginationParams = {};
 
     const [
-      { regularTransactions, nextPageParams },
+      { regularTransactions, nextPageUrl: newNextPageUrl },
       pendingTransactions,
       tipWithdrawnTransactions,
     ] = await Promise.all([
-      this.fetchTransactionsFromMiddleware(address, nextPage, TXS_PER_PAGE),
+      this.fetchRegularTransactions(address, TXS_PER_PAGE, nextPageUrl),
       this.fetchPendingTransactions(address),
-      this.fetchTipWithdrawnTransactions(address, !nextPage),
+      this.fetchTipWithdrawnTransactions(address, !nextPageUrl),
     ]);
 
+    if (newNextPageUrl) {
+      paginationParams.nextPageUrl = newNextPageUrl;
+    }
+
     return {
       regularTransactions,
-      nextPageParams,
       pendingTransactions,
       tipWithdrawnTransactions,
+      paginationParams,
+    };
+  }
+
+  override async fetchAccountAssetTransactions(
+    address: AccountAddress,
+    assetContractId: AssetContractId,
+    { nextPageUrl }: ITransactionApiPaginationParams = {},
+  ): Promise<IFetchTransactionResult> {
+    let paginationParams: ITransactionApiPaginationParams = {};
+    let regularTransactions: ITransaction[] = [];
+
+    try {
+      // Aeternity protocol APIs has no way for fetching the AE Coin transactions only.
+      // For this purpose we are fetching all transactions and filter the coin.
+      if (assetContractId === this.coinContractId) {
+        const res = await this.fetchAccountTransactions(address, { nextPageUrl });
+        paginationParams = res.paginationParams;
+        regularTransactions = res.regularTransactions
+          .filter(({ tx }) => !tx.contractId || tx.contractId === this.coinContractId);
+      } else {
+        const { getMiddleware, fetchFromMiddlewareCamelCased } = useMiddleware();
+        const middleware = await getMiddleware();
+        /**
+         * TODO `getTxs` does not provide incoming token transactions
+         * @see https://github.com/aeternity/ae_mdw/issues/1678
+         */
```

Comment:
@CedrikNikita @martinkaintas what do you think about leaving this for now and fixing this in a separate task? Maybe MDW team will provide us nice endpoint that would solve the issue.

Reply:
I suppose we can leave it for now, but if there's no response from the MDW we should find a workaround for this

Reply:
Not applicable for now. Sotiris requested to go back and use the `activities/` endpoint with all drawbacks that comes from this.